delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/client/S3StorageClient.scala

```diff
@@ -28,12 +34,22 @@ trait S3StorageClient {
 
   def headObject(bucket: String, key: String): IO[HeadObjectResponse]
 
-  def underlyingClient: S3AsyncClientOp[IO]
+  def uploadFile(
+      fileData: Stream[IO, Byte],
+      bucket: String,
+      key: String,
+      algorithm: DigestAlgorithm
+  ): IO[UploadMetadata]
+
+  def objectExists(bucket: String, key: String): IO[Boolean]
```

Comment:
Cleaner 👍  could also be a `final def` and written in terms of `headObject` on the interface